docs/guides/software-lifecycle/continuous-testing/testing-frameworks.md

```diff
@@ -3,119 +3,172 @@ title: Testing Frameworks
+            - [NSubstitute](https://nsubstitute.github.io/): A friendly substitute for .NET mocking libraries.
+            - [FakeItEasy](https://fakeiteasy.github.io/): An easy-to-use mocking framework for .NET.
+         - **C/C++**
+            - [CMock](https://github.com/ThrowTheSwitch/CMock): A mocking framework for C.
+            - [cpp-stub](https://github.com/coolxv/cpp-stub): A C++ mocking framework.
+            - [googletest](https://github.com/google/googletest): The Google C++ Testing Framework.
+         - **HTTP Services**
+            - [WireMock](http://wiremock.org/): A tool for mocking HTTP services, useful for simulating APIs and services during testing.
+      
+      
+              
+   - For Code Analysis
+      - For Static Analysis
+         - **Java**
+            - [PMD](https://github.com/pmd/pmd): A source code analyzer for Java, JavaScript, and more.
+            - [Checkstyle](https://checkstyle.sourceforge.io/): A tool for checking Java code against coding standards.
+            - [Spotbugs](https://github.com/spotbugs/spotbugs): A static analysis tool for Java bytecode.
+         - **JavaScrpt/TypeScript**
+            - [ESLint](https://eslint.org/): A widely used JavaScript and TypeScript linting tool for identifying and fixing code issues.
+            - [SonarQube](https://www.sonarqube.org/): A comprehensive platform for continuous inspection of code quality and security.
+            - [PMD](https://github.com/pmd/pmd): A source code analyzer for Java, JavaScript, and more.
+         - **Python**
+            - [PyLint](https://pylint.pycqa.org/en/latest/): A Python static code analysis tool.
+            - [Pyflakes](https://github.com/PyCQA/pyflakes): A lightweight Python code checker.
+         - **C#**
+            - [SonarQube](https://www.sonarqube.org/): A comprehensive platform for continuous inspection of code quality and security.
+         - **C/C++**
+            - [CppCheck](https://github.com/danmar/cppcheck): A static analysis tool for C/C++.
+            - [Clang Static Analyzer](https://clang-analyzer.llvm.org/) / [tidy](https://clang.llvm.org/extra/clang-tidy/): Analyzers for C/C++ code.
+
+      - For Dynamic Analysis
+         - For Test Coverage
+            - **Java**
+               - [Jacoco](https://www.jacoco.org/jacoco/): A Java code coverage library.
+               - [Cobertura](https://github.com/cobertura/cobertura): A Java code coverage tool.
+            - **Python**
+               - [Coverage](https://github.com/nedbat/coveragepy): A Python code coverage measurement tool.
+               - [PyTest-cov](https://github.com/pytest-dev/pytest-cov): A PyTest plugin for coverage reporting.
+            - **C#**
+               - [AltCover](https://github.com/SteveGilham/altcover): A .NET code coverage tool.
+               - [NCover](http://ncover.sourceforge.net/): A code coverage tool for .NET.
+            - **C/C++**
```

Comment:
CodeSonar here? We do have a reason to list it based on our implementation of the product.